lib/url_helper.rb

```diff
@@ -24,6 +24,28 @@ def self.relaxed_parse(url)
   rescue URI::Error
   end
 
+  def self.is_valid_url_or_path?(url)
+    uri = URI.parse(url)
+
+    return true if uri.is_a?(URI::Generic) && url.starts_with?("/")
+
+    if uri.scheme
+      return true if uri.is_a?(URI::MailTo)
+
+      if url.match?(%r{\A#{uri.scheme}://[^/]}) &&
+           (
+             uri.is_a?(URI::HTTP) || uri.is_a?(URI::HTTPS) || uri.is_a?(URI::FTP) ||
```

Comment:
I am wondering if the allowed schemes should also be an argument and be these 4 by default.

Reply:
Possibly but we don't have a need for that now so I don't think we should add that flexibility